test/integration/noderesourcetopology_test.go

```diff
@@ -0,0 +1,729 @@
+				},
+			},
+			PreFilter: &schedapi.PluginSet{
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+			Reserve: &schedapi.PluginSet{
+				Disabled: []schedapi.Plugin{
+					{Name: "*"},
+				},
+			},
+		},
+		PluginConfig: []schedapi.PluginConfig{
+			{
+				Name: noderesourcetopology.Name,
+				Args: args,
+			},
+		},
+	}
+
+	testCtx = util.InitTestSchedulerWithOptions(
+		t,
+		testCtx,
+		true,
+		scheduler.WithProfiles(profile),
+		scheduler.WithFrameworkOutOfTreeRegistry(registry),
+	)
+	t.Log("init scheduler success")
+	defer testutils.CleanupTest(t, testCtx)
+
+	// Create a Node.
+	nodeName1 := "fake-node-1"
+	node1 := st.MakeNode().Name("fake-node-1").Label("node", nodeName1).Obj()
+	node1.Status.Allocatable = v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewQuantity(4, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+	}
+	node1.Status.Capacity = v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewQuantity(4, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+	}
+	n1, err := cs.CoreV1().Nodes().Create(ctx, node1, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName1, err)
+	}
+
+	t.Logf(" Node 1 created: %v", n1)
+	// Create another Node.
+	nodeName2 := "fake-node-2"
+	node2 := st.MakeNode().Name("fake-node-2").Label("node", nodeName2).Obj()
+	node2.Status.Allocatable = v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewQuantity(4, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+	}
+	node2.Status.Capacity = v1.ResourceList{
+		v1.ResourceCPU:    *resource.NewQuantity(4, resource.DecimalSI),
+		v1.ResourceMemory: *resource.NewQuantity(100, resource.DecimalSI),
+		v1.ResourcePods:   *resource.NewQuantity(32, resource.DecimalSI),
+	}
+	n2, err := cs.CoreV1().Nodes().Create(ctx, node2, metav1.CreateOptions{})
+	if err != nil {
+		t.Fatalf("Failed to create Node %q: %v", nodeName2, err)
+	}
+	t.Logf(" Node 2 created: %v", n2)
+	nodeList, err := cs.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	t.Logf(" NodeList: %v", nodeList)
+	pause := imageutils.GetPauseImageName()
+	for _, tt := range []struct {
+		name                   string
+		pods                   []*v1.Pod
+		nodeResourceTopologies []*topologyv1alpha1.NodeResourceTopology
+		expectedNodes          []string
+	}{
+		{
+			name: "Filtering out nodes that cannot fit resources on a single numa node in case of Guaranteed pod",
+			pods: []*v1.Pod{
+				withContainer(st.MakePod().Namespace(ns.Name).Name("topology-aware-scheduler-pod").Req(map[v1.ResourceName]string{v1.ResourceCPU: "4", v1.ResourceMemory: "5Gi"}).Obj(), pause, withResList("4", "50Gi"), withResList("4", "50Gi")),
```

Comment:
One thing I don't like in this code: we need to specify resources twice because Req() just fills Requests, but not Limits.

I added PR for it https://github.com/kubernetes/kubernetes/pull/100658
of course we can introduce our own local helper.

Reply:
Sure, but unfortunately one more thing we can't do even with https://github.com/kubernetes/kubernetes/pull/100658 is to be able to define pods of different QoS classes. With your addition of Limit in the PodWrapper we would only able to create Guaranteed pods and still have to create burstable/BE pods using the local helper.  It might worth investing some time to improve the pod wrapper to be able to define pods with various QoS with inputs of request and limit. Adding a similar comment in the PR as well.